### PR TITLE
Prevent flashing of component

### PR DESCRIPTION
**Pull request by karland**
The component was still flashing, so I read up:

1. setState() is asnychronous. 
[https://reactjs.org/docs/react-component.html#setstate](https://reactjs.org/docs/react-component.html#setstate)

2. do not use componentWillMount():
[https://reactjs.org/docs/react-component.html#unsafe_componentwillmount](https://reactjs.org/docs/react-component.html#unsafe_componentwillmount)
    >UNSAFE_componentWillMount() is invoked just before mounting occurs. It is called before render(), therefore calling setState() synchronously in this method will not trigger an extra rendering. Generally, we recommend using the constructor() instead for initializing state.

    >Avoid introducing any side-effects or subscriptions in this method. For those use cases, use componentDidMount() instead.

3. the correct place is componentDidMount():
[https://reactjs.org/docs/react-component.html#componentdidmount](https://reactjs.org/docs/react-component.html#componentdidmount):

    >componentDidMount() is invoked immediately after a component is mounted. Initialization that requires DOM nodes should go here. If you need to load data from a remote endpoint, this is a good place to instantiate the network request.
    > [...]
    >Calling setState() in this method will trigger an extra rendering, but it will happen before the browser updates the screen. This guarantees that even though the render() will be called twice in this case, the user won’t see the intermediate state.

    I suppose, loading a cookie is remote.

### CODE REVIEW

**Comment by Mastermindzh** [comment]
I will accept the pull-request but not for the reasons stated.

ComponentWillMount is bound to be deprecated (hence UNSAFE) in the new React version so switching to componentDidMount would be a solid plan. 

The documentation is spot on about a few things:

>  Generally, we recommend using the constructor() instead for initializing state.

This is what we're doing already ([we don't have a setState but rather a this.state = {}](https://github.com/Mastermindzh/react-cookie-consent/blob/master/src/index.js#L15-L16)), we're initializing the component with a visiblity of "false".

This means that, regardless of component will/did mount there shouldn't ever be a render [since we're returning null](https://github.com/Mastermindzh/react-cookie-consent/blob/master/src/index.js#L67-L69).

In short, I don't think that this is what is causing the flashing for you, if you make part of your code available (gist maybe?) I'd be more than happy to have a look! We might be able to figure out why it's flashing after all!

**Comment by karland** [comment]
Thanks for merging. The flashing of the bar is now gone. I am happy to share my code, but there is nothing to share really. It's all from out of the box. I am using Gatsby for a website, which employs a layout/index.js:

```jsx
const TemplateWrapper = ({ children }) => (
  <div>
    <CookieConsent>
      This website uses cookies for administrative purposes and to enhance the
      user experience. By using this website you agree to their application.
    </CookieConsent>
    <DCHeader />
    <Container>{children()}</Container>
    <DCFooter />
  </div>
)
```

So with this straight forward code, I was wondering what was going on and why I saw the flashing of the bar for a fraction of a second. 

The only explanation I could come up with was that it had something to do with the setState(). But, in the meanwhile I am convinced that your reasoning is correct and it was not the code. 

I now think it is a webpack issue. I did not see the flashing while doing `gatsby develop` but only after carrying out a `gatsby build`. 

With moving from componentWillMount() to componentDidMount() the issue is fixed and the code now is also future proof. Thanks.
